### PR TITLE
feat: bitnami/external-dns: Support for Azure workload Identity

### LINKED ISSUE #19141: [bitnami/external-dns] Add azure workload identity support
### Name and Version

bitnami/external-dns

### What is the problem this feature will solve?

In external-dns v0.13.6 azure workload identity support was added and the chart needs to be updated to add support

### What is the feature you are proposing to solve the problem?

To add support for workload identity, the implementation is similar to the existing `useManagedIdentityExtension` option as the ability to set `useWorkloadIdentityExtension` in the json config file needs to be available which can be seen here: https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/azure.md#helm

Full instructions: https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/azure.md#managed-identity-using-workload-identity

### What alternatives have you considered?

Manually patching the helm chart can work but is clunky and since the current implementation of pod/managed identity is being removed at the end of September, adding native support for this new authentication configuration is important.

### PR DESCRIPTION
**Pull request by michelefa1988**
### Description of the change

Azure aad-pod-identity has now been depreciated. It has been replaced with [Azure workload identity]([Managed Identity Using Workload Identity](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/azure.md#managed-identity-using-workload-identity)). External DNS docker image has now been updated and it now supports workload identity but the helm chart does not support it. This PR adds support for the missing feature

### Benefits

Support for a new essential feature. 

### Possible drawbacks

None that I know of 

### Applicable issues

- fixes #19141 

### Additional information

Test which I ran 
```yaml
## ServiceAccount parameters
## https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/
##
serviceAccount:
  ## @param serviceAccount.create Determine whether a Service Account should be created or it should reuse a exiting one.
  ##
  create: false
  ## @param serviceAccount.name ServiceAccount to use. A name is generated using the external-dns.fullname template if it is not set
  ##
  name: "external-dns-public"
  ## @param serviceAccount.annotations Additional Service Account annotations
  ##
  annotations: {}
  ## @param serviceAccount.automountServiceAccountToken Automount API credentials for a service account.
  ##
  automountServiceAccountToken: true
  ## @param serviceAccount.labels [object] Additional labels to be included on the service account
  ##
  labels: 
    azure.workload.identity/use: "true"

podLabels: 
  azure.workload.identity/use: "true"

provider: `azure`

azure:
  ## When a secret to load azure.json is not specified, the host's /etc/kubernetes/azure.json will be used
  ## @param azure.secretName When using the Azure provider, set the secret containing the `azure.json` file
  ##
  secretName: ""
  ## @param azure.cloud When using the Azure provider, set the Azure Cloud
  ##
  cloud: ""
  ## @param azure.resourceGroup When using the Azure provider, set the Azure Resource Group
  ##
  resourceGroup: "rg-xxx-xxxx-xxx"
  ## @param azure.tenantId When using the Azure provider, set the Azure Tenant ID
  ##
  tenantId: "xxxx-xxx-xxxx-xxx-xxxxx"
  ## @param azure.subscriptionId When using the Azure provider, set the Azure Subscription ID
  ##
  subscriptionId: "xxxx-xxxx-xxx-xxx-xxxxx"
  ## @param azure.aadClientId When using the Azure provider, set the Azure AAD Client ID
  ##
  aadClientId: ""
  ## @param azure.aadClientSecret When using the Azure provider, set the Azure AAD Client Secret
  ##
  aadClientSecret: ""
  ## @param azure.useWorkloadIdentityExtension When using the Azure provider, set if you use Workload Identity
  ##
  useWorkloadIdentityExtension: true
  ## @param azure.userAssignedIdentityID When using the Azure provider with Azure MSI, set Client ID of Azure user-assigned managed identity (optional, otherwise system-assigned managed identity is used)
  ##
  useManagedIdentityExtension: false
  ## @param azure.userAssignedIdentityID When using the Azure provider with Azure MSI, set Client ID of Azure user-assigned managed identity (optional, otherwise system-assigned managed identity is used)
  ##
  userAssignedIdentityID: ""

```
values.yaml for provider `azure-private-dns`
```yaml
## ServiceAccount parameters
## https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/
##
serviceAccount:
  ## @param serviceAccount.create Determine whether a Service Account should be created or it should reuse a exiting one.
  ##
  create: false
  ## @param serviceAccount.name ServiceAccount to use. A name is generated using the external-dns.fullname template if it is not set
  ##
  name: "external-dns-private"
  ## @param serviceAccount.annotations Additional Service Account annotations
  ##
  annotations: {}
  ## @param serviceAccount.automountServiceAccountToken Automount API credentials for a service account.
  ##
  automountServiceAccountToken: true
  ## @param serviceAccount.labels [object] Additional labels to be included on the service account
  ##
  labels: 
    azure.workload.identity/use: "true"

podLabels: 
  azure.workload.identity/use: "true"

provider: azure-private-dns


azure:
  ## When a secret to load azure.json is not specified, the host's /etc/kubernetes/azure.json will be used
  ## @param azure.secretName When using the Azure provider, set the secret containing the `azure.json` file
  ##
  secretName: ""
  ## @param azure.cloud When using the Azure provider, set the Azure Cloud
  ##
  cloud: ""
  ## @param azure.resourceGroup When using the Azure provider, set the Azure Resource Group
  ##
  resourceGroup: "rg-xxxx-xxxx-xxxx"
  ## @param azure.tenantId When using the Azure provider, set the Azure Tenant ID
  ##
  tenantId: "xxx-xxxx-xxxx-xxxx-xxxx"
  ## @param azure.subscriptionId When using the Azure provider, set the Azure Subscription ID
  ##
  subscriptionId: "xxxx-xxxx-xxxx-xxxx-xxxx"
  ## @param azure.aadClientId When using the Azure provider, set the Azure AAD Client ID
  ##
  aadClientId: ""
  ## @param azure.aadClientSecret When using the Azure provider, set the Azure AAD Client Secret
  ##
  aadClientSecret: ""
  ## @param azure.useWorkloadIdentityExtension When using the Azure provider, set if you use Workload Identity
  ##
  useWorkloadIdentityExtension: true
  ## @param azure.userAssignedIdentityID When using the Azure provider with Azure MSI, set Client ID of Azure user-assigned managed identity (optional, otherwise system-assigned managed identity is used)
  ##
  useManagedIdentityExtension: false
  ## @param azure.userAssignedIdentityID When using the Azure provider with Azure MSI, set Client ID of Azure user-assigned managed identity (optional, otherwise system-assigned managed identity is used)
  ##
  userAssignedIdentityID: ""

```

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [ ] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [ ] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [ ] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/external-dns/values.yaml [hunk 1]
```diff
@@ -276,7 +276,10 @@ azure:
   ## @param azure.aadClientSecret When using the Azure provider, set the Azure AAD Client Secret
   ##
   aadClientSecret: ""
-  ## @param azure.useManagedIdentityExtension When using the Azure provider, set if you use Azure MSI
+  ## @param azure.useWorkloadIdentityExtension When using the Azure provider, set if you use Workload Identity
+  ##
+  useWorkloadIdentityExtension: false
+  ## @param azure.userAssignedIdentityID When using the Azure provider with Azure MSI, set Client ID of Azure user-assigned managed identity (optional, otherwise system-assigned managed identity is used)
```
Comment (carrodher) [inline bitnami/external-dns/values.yaml:282]: Thanks for your contribution. Please note here you need to add the documentation about the recently added `useManagedIdentityExtension: false` parameter

Comment (michelefa1988) [reply bitnami/external-dns/values.yaml:282]: Not sure I understood where I have to place the documentation. I have added some extra comments in the values.yaml to help guide people

Comment (michelefa1988) [reply bitnami/external-dns/values.yaml]: I think I maaged

**Review by juan131 — APPROVED** [review]
Thanks for this contribution! LGTM

**Pull request merged**